pkg/openstack/loadbalancer_sg.go

```diff
@@ -75,6 +75,23 @@ func applyNodeSecurityGroupIDForLB(network *gophercloud.ServiceClient, svcConf *
 				continue
 			}
 
+			if port.DeviceID != serverID {
```

Comment:
should we have a check similar to line 101?

Reply:
Good point. I actually think we can move the new block after L101, which will help to leave the loop earlier if the port isn't part of the LB's subnet.